### ISSUE TITLE
dbplyr 2.4.0 fails to read CSV from Spark with regexpr error

### ISSUE THREAD

**Issue by tomshafer** [tidyverse/dbplyr#1387]
In conjunction with **sparklyr**:

**dbplyr 2.4.0:**

```r
sc <- sparklyr::spark_connect(method = "databricks")
sparklyr::spark_read_csv(sc, path)
```

```
Warning: `src_sql()` was deprecated in dbplyr 1.4.0.
ℹ The deprecated feature was likely used in the sparklyr package.
  Please report the issue at <https://github.com/sparklyr/sparklyr/issues>.
Error in purrr::map(clauses, sql_format_clause, lvl = lvl, con = con) : 
  ℹ In index: 2.
Caused by error in `` regexpr("`", x)[[1]] ``:
! subscript out of bounds
```

**dbplyr 2.3.4:**

```r
remotes::install_version("dbplyr", version = "2.3.4")

sc <- sparklyr::spark_connect(method = "databricks")
sparklyr::spark_read_csv(sc, path)
```

(Works fine.)

**Comment by hadley** [comment]
cc @edgararuiz

**Comment by edgararuiz** [comment]
Hi @tomshafer , what version of `sparklyr` are you running, I believe that was addressed in version 1.8.4

**Comment by tomshafer** [comment]
Thanks, @edgararuiz! At the time I think it was 1.8.3 as the newer one wasn’t available from CRAN. All seems OK now, with 1.8.4 installed, so I’ll close the issue.
